python/lsst/ts/pre_commit_conf/pre_commit_hooks.py

```diff
@@ -0,0 +1,198 @@
+# This file is part of ts_pre_commit_conf.
+#
+# Developed for the Vera Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["registry"]
+
+from dataclasses import dataclass
+
+
+@dataclass(eq=True, frozen=True)
+class PreCommitHookMetadata:
+    """Dataclass representing all metadata for a pre-commit hook.
+
+    Attributes
+    ----------
+    config_file_name : `str` | None
+        The name of the configuration file. Set to None if no configuration
+        file exists for this hook.
+    config : `str` | None
+        The configuration of the pre-commit hook as contained by the
+        configuration file. Set to None if no configuration file exists for
+        this hook.
+    pre_commit_config : `str`
+        The YAML string to add to the pre-commit-config.yaml file.
+    optional : `bool`
```

Comment:
You can save some lines by providing default values, such as:

   excludable: bool = False
   
Then you can drop executable = False from constructor calls.